src/sentry/migrations/0748_fix_substatus_for_ignored_groups.py

```diff
@@ -0,0 +1,107 @@
+# Generated by Django 5.0.7 on 2024-08-05 17:50
+
+from django.apps.registry import Apps
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+
+from sentry.new_migrations.migrations import CheckedMigration
+
+# Copying constants defined in the models
+
+
+class ActivityType:
+    SET_IGNORED = 3
+
+
+class GroupHistoryStatus:
+    ARCHIVED_UNTIL_ESCALATING = 15
+    ARCHIVED_FOREVER = 16
+    ARCHIVED_UNTIL_CONDITION_MET = 17
+
+
+class GroupSubStatus:
+    UNTIL_ESCALATING = 1
+    # Group is ignored/archived for a count/user count/duration
+    UNTIL_CONDITION_MET = 4
+    # Group is ignored/archived forever
+    FOREVER = 5
+
+
+class GroupStatus:
+    IGNORED = 2
+
+
+# End copy
+
+ACTIVITY_DATA_FIELDS = {
+    "ignoreCount",
+    "ignoreDuration",
+    "ignoreUntil",
+    "ignoreUserCount",
+    "ignoreUserWindow",
+    "ignoreWindow",
+}
+
+
+def backfill_substatus_for_ignored_groups(
+    apps: Apps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    Group = apps.get_model("sentry", "Group")
+    Activity = apps.get_model("sentry", "Activity")
+    GroupSnooze = apps.get_model("sentry", "GroupSnooze")
+
+    groups = Group.objects.filter(status=GroupStatus.IGNORED, substatus=None)
```

Comment:
Do we need to use any RangeQuerySetWrapper?
https://develop.sentry.dev/development/database-migrations/#filters

What happens if the migration fails? Does this get scheduled outside of a normal deployment? (I have not done migrations in such a long time)

Should we use the new GoCD script to run this update?

Reply:
No script as we should run this for self-hosted as well (Dan explained this to me)

Reply:
I was hoping to avoid using `RangeQuerySetWrapper` since we wouldn't want to pair that with the `.filter` on substatus. 

This will be run manually with the GoCD migrations job, and is safe to rerun if it fails.